bundles/ranvier-combat/player-events.js

```diff
@@ -316,28 +317,70 @@ module.exports = (srcPath) => {
           B.sayAt(this.party, `<b><green>${this.name} was killed!</green></b>`);
         }
 
-        this.setAttributeToMax('physical');
+        B.sayAt(this, '<b><red>You have died. You feel the last of your essence slipping from this mortal husk.</red></b>');
 
-        let home = state.RoomManager.getRoom(this.getMeta('waypoint.home'));
-        if (!home) {
-          home = state.RoomManager.startingRoom;
+        const items = [];
+
+        if (this.inventory) {
+          items.push(...this.inventory.values());
+        }
+
+        if (this.equipment) {
+          items.push(...this.equipment.values());
         }
 
-        this.moveTo(home, _ => {
-          state.CommandManager.get('look').execute(null, this);
+        // Make and drop a corpse w/ player inventory/equipment.
+        const corpse = new Item(this.area, {
+          id: 'corpse',
+          name: `Corpse of ${this.name}`,
+          roomDesc: `Corpse of ${this.name}`,
+          description: `The rotting corpse of ${this.name}`,
+          keywords: ['corpse', this.name],
+          type: 'CONTAINER',
+          properties: {
+            noPickup: true,
+          },
+          maxItems: items.length + 1,
+          behaviors: {
+            decay: {
+              duration: 600
```

Comment:
Might add a check to set this to a very low number for level 1 or 2 characters to prevent abuse.